azure-postgresql-flexible-server.yml

```diff
@@ -0,0 +1,192 @@
+# Copyright 2018 the Service Broker Project Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+version: 1
+name: csb-azure-postgresql-flexible-server
+id: d69dd4aa-e27e-490c-bdbf-c887563da27f
+description: Azure Database for PostgreSQL - flexible server
+display_name: Azure Database for PostgreSQL - flexible server
+image_url: file://service-images/csb.png
+documentation_url: https://learn.microsoft.com/en-gb/azure/postgresql/
+support_url: https://learn.microsoft.com/en-gb/azure/postgresql/
+tags: [azure, postgresql, postgres, preview, flexible server]
+plan_updateable: true
+provision:
+  plan_inputs: []
+  user_inputs:
+  - field_name: postgres_version
+    type: string
+    details: The version for the PostgreSQL instance.
+    default: "11"
+  - field_name: storage_gb
+    type: number
+    details: Size of storage volume for service instance.
+    default: 32
+    constraints:
+      maximum: 4096
+      minimum: 5
+  - field_name: instance_name
+    type: string
+    details: Name for your PostgreSQL instance
+    default: csb-postgresql-${request.instance_id}
+    constraints:
+      maxLength: 98
+      minLength: 6
+      pattern: ^[a-z][a-z0-9-]+$
+    prohibit_update: true
+  - field_name: resource_group
+    type: string
+    details: Name for the resource group for this instance
+    default: ''
+    constraints:
+      maxLength: 64
+      minLength: 0
+      pattern: ^[a-z][a-z0-9-]+$|^$
+    prohibit_update: true
+  - field_name: db_name
+    type: string
+    details: Name for your database
+    default: vsbdb
+    constraints:
+      maxLength: 64
+  - field_name: delegated_subnet_id
+    type: string
+    details: |
+      The ID of the virtual network subnet to create the PostgreSQL Flexible Server. 
+      The provided subnet should not have any other resource deployed in it and this subnet will be delegated to the PostgreSQL Flexible Server, if not already delegated.
+    nullable: true
```

Comment:
Transitionally we have used `nullable` where `""` was one of the valid values (and therefore couldn't be interpreted as the unset value). Is this consistent with other usages? Could it be over-engineering? Is is consistent with `private_endpoint_subnet_id` whose default is `""`?

Reply:
I think is set it to nullable because empty string was being interpreted as trying to set it to a value - I will test again and confirm.

Reply:
I confirmed that passing `delegated_subnet_id = ""` results in an error 
```
"error":"Error: parsing \"\": cannot parse an empty string with azurerm_postgresql_flexible_server.instance, on main.tf line 54, in resource \"azurerm_postgresql_flexible_server\" \"instance\": 54:   delegated_subnet_id          = \"\" exit status 1","message":"provision failed: Error: parsing \"\": cannot parse an empty string with azurerm_postgresql_flexible_server.instance, on main.tf line 54, in resource \"azurerm_postgresql_flexible_server\" \"instance\": 54:   delegated_subnet_id          = \"\" exit status 1"}}

```

Reply:
The same error occurs for the `private_dns_zone_id` property so they both need to be nullable